xonsh/main.py

```diff
@@ -224,8 +224,8 @@ def main(argv=None):
         # run a script contained in a file
         path = os.path.abspath(os.path.expanduser(args.file))
         if os.path.isfile(path):
-            sys.argv = args.args
-            env['ARGS'] = [args.file] + args.args
+            sys.argv = [args.file] + args.args
+            env['ARGS'] = sys.argv[:]  # $ARGS is not sys.argv
```

Comment:
you're right, if you reassign $ARGS to a new list then it doesn't matter anyway but if you mutate in place then it is a feature more than a bug imo.

> 
The very fact that $ARGS += something works but $ARGS = $ARGS + something doesn't is, in my opinion, extremely counter-intuitive.

that is a python implementation detail, in-place addition uses `list.extend()` while the second one creates a new list.